modules/augment/include/opencv2/augment/augmenter.hpp

```diff
@@ -0,0 +1,29 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+
+#ifndef OPENCV_AUGMENT_AUGMENTER_HPP
+#define OPENCV_AUGMENT_AUGMENTER_HPP
+#include <opencv2/augment/transform.hpp>
+#include <tuple>
+namespace cv { namespace augment {
+
+class CV_EXPORTS_W Augmenter
+{
+public:
+    /* @brief Constructor
+    */
+    CV_WRAP Augmenter();
+
+    /* @brief add a transformation to the augmenter
+    */
+    CV_WRAP void add(Ptr<Transform> t, float prob=1.f);
```

Comment:
Please start adding tests in this PR. At least just for API (try to reproduce Augmenter creation, transformations collecting, applying to images/ground truth/batch of images).

Reply:
I think I will add test cases to this PR after finishing the implementation of the augmenter class, as it can be used to test all the implemented transformations.. do you agree?

also I think reproduction should be easy since we use rng for all random number generations